### PR TITLE
Small fixes to the configdir code

### PR DESCRIPTION
**Pull request by aykevl**
* Add locking so no two go-librespot processes can use the same configdir
* Use indentation in state.json
* Hopefully fix some atomicity issues when overwriting the state file

See also the discussion in #103.

### CODE REVIEW

File: cmd/daemon/main.go [hunk 1]
```diff
@@ -397,6 +406,25 @@ func loadConfig(cfg *Config) error {
...
+	cfg.configLock = flock.New(lockFilePath)
+	locked, err := cfg.configLock.TryLock()
+	if err != nil {
+		return fmt.Errorf("could not lock config directory: %w", err)
+	}
+	if !locked {
+		// Lock already taken! Looks like go-librespot is already running.
+		return fmt.Errorf("%w (lockfile: %s)", errAlreadyRunning, lockFilePath)
+	}
+
```
Comment (devgianlu) [inline cmd/daemon/main.go:427]: This could be compacted to a single block so that `locked` does not clobber. Just nitpicking...

**Review by devgianlu — APPROVED** [review]

File: cmd/daemon/main.go (hunk 1 above)
Comment (aykevl) [reply cmd/daemon/main.go:427]: You mean something like this?

```go
	if locked, err := cfg.configLock.TryLock(); err != nil {
		return fmt.Errorf("could not lock config directory: %w", err)
	} else if !locked {
		// Lock already taken! Looks like go-librespot is already running.
		return fmt.Errorf("%w (lockfile: %s)", errAlreadyRunning, lockFilePath)
	}
```

I guess, yeah. I'm not sure it's more readable but I'm fine either way.

Comment (devgianlu) [reply cmd/daemon/main.go:427]: Yeah... Just not to have `locked` in the scope

Comment (aykevl) [reply cmd/daemon/main.go]: Fixed!

**Review by devgianlu — APPROVED** [review]

**Pull request merged**